packages/transaction-controller/src/utils/batch.ts

```diff
@@ -106,24 +106,37 @@ export const ERROR_MESSAGE_NO_UPGRADE_CONTRACT =
 export async function addTransactionBatch(
   request: AddTransactionBatchRequest,
 ): Promise<TransactionBatchResult> {
-  const { getInternalAccounts, messenger, request: userRequest } = request;
+  const {
+    getInternalAccounts,
+    messenger,
+    request: transactionBatchRequest,
+  } = request;
   const sizeLimit = getBatchSizeLimit(messenger);
 
   validateBatchRequest({
     internalAccounts: getInternalAccounts(),
-    request: userRequest,
+    request: transactionBatchRequest,
     sizeLimit,
   });
 
-  const { useHook } = userRequest;
-
-  log('Adding', userRequest);
-
-  if (useHook) {
-    return await addTransactionBatchWithHook(request);
+  log('Adding', transactionBatchRequest);
+
+  if (!transactionBatchRequest.disable7702) {
+    try {
+      return await addTransactionBatchWith7702(request);
+    } catch (error: unknown) {
+      if (
+        error instanceof JsonRpcError &&
+        error.message === 'Chain does not support EIP-7702'
+      ) {
+        log('Falling back to hook-based batch processing');
+        return await addTransactionBatchWithHook(request);
```

Comment:
Minor, could we invert the `if` condition to only throw if not this, then we wouldn't need to duplicate this and it would continue on below.